src/Client.php

```diff
@@ -0,0 +1,258 @@
+        }
+
+        return $responseData;
+    }
+
+    public function setHttpClientFactory(HttpClientFactory $httpClientFactory)
+    {
+        $this->httpClientFactory = $httpClientFactory;
+    }
+
+    public function fetchAccountBalance(string $accountNumber): FetchAccountBalanceResponse
+    {
+        $uri = $this->config->getApiBaseUrl() . '/v2/transact';
+
+        $request = $this->requestFactory->createRequest('GET', $uri)
+            ->withHeader('Authorization', 'Bearer ' . $this->config->getApiKey())
+            ->withHeader('Content-Type', 'application/json')
+            ->withHeader('Accept', 'application/json')
+            ->withQueryParams(['account_number' => $accountNumber]);
+
+        $response = $this->httpClient->sendRequest($request);
+        $responseData = json_decode($response->getBody()->getContents(), false);
+
+        return new FetchAccountBalanceResponse(
+            $responseData->status,
+            $responseData->message,
+            $responseData->data
+        );
+    }
+
+    public function fetchDomesticBankAccountNameRaw(string $requestRef, string $accountNumber): ResponseInterface
+    {
+        $url = $this->config->getBaseUrl() . '/retail/account/resolve/account-number';
+        $headers = $this->getRequestHeaders();
+        $payload = json_encode([
+            'requestRef' => $requestRef,
+            'accountNumber' => $accountNumber,
+        ]);
+
+        // Make the HTTP request and get the response
+        $httpClient = $this->getHttpClient();
+        $response = $httpClient->request(HttpMethodEnum::POST, $url, [
+            'headers' => $headers,
+            'body' => $payload,
+        ]);
+
+        // Return the response object
+        return new Response(
+            $response->getStatusCode(),
+            $response->getHeaders(),
+            $response->getBody(),
+            $response->getProtocolVersion(),
+            $response->getReasonPhrase()
+        );
+    }
+
+    public function fetchDomesticTransactionStatusRaw(BankTransactionInterface $bankTransaction): TransactionResponse
+    {
+        if($bankTransaction instanceof SourceModelInterface){
+            $this->setSourceModel($bankTransaction);
+        }
+
+        $response = $this->performRequest(HttpMethodEnum::POST, 'bankAccountFT', [
+            'amount' => $bankTransaction->getAmount(),
+            'destination_account' => $bankTransaction->getDestinationAccount(),
+            'destination_bank_code' => $bankTransaction->getDestinationBankCode(),
+            'request_ref' => $bankTransaction->getRequestRef(),
+            'transaction_ref' => $bankTransaction->getTransactionRef(),
+            'description' => $bankTransaction->getTransactionDesc(),
+        ]);
+
+        return new TransactionResponse($response);
+    }
+
+    public function sendDomesticTransaction(BankTransactionInterface $bankTransaction): TransactionResponse
```

Comment:
please add the doc block to all methods with the link to the polaris API documentation like we agreed
example:

```suggestion
    /** @link link-to-the-api-method-documentation */
    public function sendDomesticTransaction(BankTransactionInterface $bankTransaction): TransactionResponse
```